diff --git a/groupmenu.info.yml b/groupmenu.info.yml
index e33a63f..7ab2a3a 100644
--- a/groupmenu.info.yml
+++ b/groupmenu.info.yml
@@ -1,4 +1,5 @@
 name: 'Group Menu'
+core_version_requirement: ^8 || ^9
 description: 'Gives the ability to create and manage menus for groups.'
 package: 'Group'
 type: 'module'
diff --git a/modules/groupmenu_block/groupmenu_block.info.yml b/modules/groupmenu_block/groupmenu_block.info.yml
index e7f90f8..666f663 100644
--- a/modules/groupmenu_block/groupmenu_block.info.yml
+++ b/modules/groupmenu_block/groupmenu_block.info.yml
@@ -1,4 +1,5 @@
 name: 'Group Menu Block'
+core_version_requirement: ^8 || ^9
 description: 'Provides a block to display the group menus for content in a group.'
 package: 'Group'
 type: 'module'
diff --git a/src/Controller/GroupMenuController.php b/src/Controller/GroupMenuController.php
index 7e5a242..561a148 100644
--- a/src/Controller/GroupMenuController.php
+++ b/src/Controller/GroupMenuController.php
@@ -8,7 +8,7 @@ use Drupal\Core\Render\RendererInterface;
 use Drupal\group\Entity\Controller\GroupContentController;
 use Drupal\group\Entity\GroupInterface;
 use Drupal\group\Plugin\GroupContentEnablerManagerInterface;
-use Drupal\user\PrivateTempStoreFactory;
+use Drupal\Core\TempStore\PrivateTempStoreFactory;
 use Symfony\Component\DependencyInjection\ContainerInterface;
 
 /**
@@ -28,7 +28,7 @@ class GroupMenuController extends GroupContentController {
    *
    * @param \Drupal\group\Plugin\GroupContentEnablerManagerInterface $plugin_manager
    *   The group content plugin manager.
-   * @param \Drupal\user\PrivateTempStoreFactory $temp_store_factory
+   * @param \Drupal\Core\TempStore\PrivateTempStoreFactory $temp_store_factory
    *   The private store factory.
    * @param \Drupal\Core\Entity\EntityTypeManagerInterface $entity_type_manager
    *   The entity type manager.
@@ -48,7 +48,7 @@ class GroupMenuController extends GroupContentController {
   public static function create(ContainerInterface $container) {
     return new static(
       $container->get('plugin.manager.group_content_enabler'),
-      $container->get('user.private_tempstore'),
+      $container->get('tempstore.private'),
       $container->get('entity_type.manager'),
       $container->get('entity.form_builder'),
       $container->get('renderer')
diff --git a/src/Plugin/GroupContentEnabler/GroupMenu.php b/src/Plugin/GroupContentEnabler/GroupMenu.php
index 1e32d57..50f4d42 100644
--- a/src/Plugin/GroupContentEnabler/GroupMenu.php
+++ b/src/Plugin/GroupContentEnabler/GroupMenu.php
@@ -59,7 +59,7 @@ class GroupMenu extends GroupContentEnablerBase {
     // and target entity type name in their titles and descriptions.
     $t_args = [
       '%plugin_name' => $this->getLabel(),
-      '%entity_type' => $this->getEntityType()->getLowercaseLabel(),
+      '%entity_type' => $this->getEntityType()->getSingularLabel(),
     ];
     $defaults = ['title_args' => $t_args, 'description_args' => $t_args];
 
